### ISSUE TITLE
request: single template library for multisite network

### ISSUE THREAD

**Issue by matthiaswagnerwy** [pojome/elementor#1225]
hy,
i guess it would be a nice feature to have a setting that changes the behaviour how templates are saved in the own library on multisite, so that every site could access all templates.
or even better, if the library detects if it is run inside a multisite and gives the possibility to choose if it is accessible only for the active site or the whole network.

**Comment by luizeof** [comment]
+1

**Comment by atMaarten** [comment]
Great idea!
Also I would be nice if new multisites are standard valided through the same license as main domain.

**Comment by ibanner** [comment]
Following @atMaarten's note (an If I understand it correctly) I guess this can be achieved either by supporting wildcard licenses for the Unlimited accounts, or by automatically adding an activation to my account when a new site is created. I think WPML behave something like this.

**Comment by rathert** [comment]
+1 on multisite wide library

**Comment by black-eye** [comment]
+1

**Comment by AdalDesign** [comment]
+1

**Comment by TheBigBanik** [comment]
+1 This is a must for every multisite. The more efficient way is to create a template and have a button "Distribute to network" where we can select All or specific subsites to apply this template. We will save HUGE amount of time.

**Comment by aidenafterall** [comment]
+1

**Comment by jhume** [comment]
+1 this would save a ton of time

**Comment by sarvesh-esokia** [comment]
Just found this plugin https://waashero.com/template-hero-for-elementor/ did any of you guys ever tried it?

**Comment by TheBigBanik** [comment]
Yep, I did purchase the Template Hero for Elementor but the only thing it does is to create templates in a different domain and wordpress installation and then those templates can be accessed by your network websites. BUT (this is the huge mistake of my purchase) if you make any changes to the original template, you have to manually reload this template to each subsite o the network, it doesn't update automatically, this is something I asked before the purchase and they said that it is possible but then I found out that it isn't. Waste of money there, just export the templates and import in any subsite, it is exactly the same.

**Comment by nerdflu** [comment]
+1 this would be a huge time saver for me. Elementor theme builder templates are a nightmare to keep sync'd across multisite networks.

**Comment by sandornemeth86** [comment]
+1 this is definitely a must have feature.

**Comment by sandornemeth86** [comment]
Hi @TheBigBanik,
I have just looked into the "Template Hero for Elementor" plugin. Are you sure that it doesn't do what we want?
I found this feature below, and it does look like that it would do what we need.
[https://docs.waashero.com/docs/template-hero-for-elementor/clone-elementor-template-from-multisite-admin-to-all-subsites/](url)

**Comment by TheBigBanik** [comment]
Hi @sandornemeth86, the way this plugin work is: 
A. Install the Server plugin in a domain
B. Install the client plugin in a multisite, different domain
C. Create in the server site the templatew with Elementor
D. Connect both Server and Client site (multisite)
E. In the client site or any subsite of your network, you can view and I.porr all templates created in the server site.
G. If you just want to create and distribute, no problem, on the other hand, you can just export and Import like before, you don't need a plugin BUT keep in mind that if you make any changes to any of the templates in the server, those changes won't reflect your multisite because it is not synchronized so you have to reinsert one by one  in every subsite. I asked them about this functionality because if you have many subsites you will need to spend a lot of time to re import all templates, and they said that they will think about it in the future in case the development of this functionality is something that they can manage to do.

**Comment by sandornemeth86** [comment]
Thanks @TheBigBanik for the details. I see. Then I won't bother going down that route. I am really looking forward to having this feature integrated into elementor at some point. Probably it will take some time though..

**Comment by nerdflu** [comment]
@sandornemeth86  Same, but this request was put in 4.5 years ago... so... I mean... I'm not waiting 4 more years 😂 sorry guys - I hope they pull the rabbit out of the hat sometime in the 21st century, but I don't think they ever will... I'm looking at either moving off multisite or moving off Elementor.. and probably both... Using them together basically quadrupled my problems... I actually miss just having a vanilla WordPress theme/child theme, the network was more stable in every way, edit the theme in one spot and it updates the entire network. Old school is the new school 😄

**Comment by cscheltinga** [comment]
As far as I know, this is still an Open issue. But I've created a workaround:
In a multisite you can use _switch_to_blog(id)_. So this way we can create a shortcode that pulls up a template from the main site:

``` 
function elem_main_template($atts) {
    $template_id = esc_attr($atts['id']);
    
    switch_to_blog(1);
    $output = do_shortcode('[elementor-template id="' . $template_id . '"]');
    restore_current_blog();

    return $output;
}
add_shortcode('main_template', 'elem_main_template');
````

**Comment by TheBigBanik** [comment]
Hi @cscheltinga, did you try that workaround and it is actually working? Have you tried that with dynamic settings? Will this work? For example (to save you time) if the main site has a custom meta field e.g. Address and the widget gets the address dynamically, will this dynamic setting work in the subsite if the same meta fields Address exists? Nice try though, I will test it.

**Comment by emtiazzahid** [comment]
thanks, @cscheltinga , seems working for me

**Comment by cscheltinga** [comment]
> Hi @cscheltinga, did you try that workaround and it is actually working? Have you tried that with dynamic settings? Will this work? For example (to save you time) if the main site has a custom meta field e.g. Address and the widget gets the address dynamically, will this dynamic setting work in the subsite if the same meta fields Address exists? Nice try though, I will test it.

Yes, tested and proven ;-). 

You might need to do some tweaking to get the widget from your main site and the meta-fields from you current site. But that should be doable.

**Comment by TheBigBanik** [comment]
Great news. Thank you for the feedback. I will test it soon.

**Comment by eingel12** [comment]
Hello @TheBigBanik !

Did you get to get the ACF from the subsite in the shortcode?

Thank you!!

**Comment by TheBigBanik** [comment]
Hi, didn't try that yet. I was very busy but I will sometime. If it works I will let everyone knows.

**Comment by eingel12** [comment]
Thank you so much @TheBigBanik. I am super interested in that issue for a project I am building. It would be fantastic if you can solve it because I couldn't 😭

**Comment by ckunderd** [comment]
This would be a game changer

**Comment by stasonua0** [comment]
+1

**Comment by stentotre** [comment]
+1

**Comment by kvzl** [comment]
+1

**Comment by iliyahristov** [comment]
+1 
Take a look here : [](https://docs.waashero.com/docs/template-hero-for-elementor/clone-elementor-template-from-multisite-admin-to-all-subsites/)
It seems that can help for now.

**Comment by mauro500designs** [comment]
+1 for multisites template shareing and global styles!

**Comment by ofmarconi** [comment]
Alguém já usou com JetEngine? Minha preocupação são os Listings e Loops.

Quais são as limitações? Os loops deveriam ser configurados como? cada loop dentro do sub-site com o shortcode?

> As far as I know, this is still an Open issue. But I've created a workaround: In a multisite you can use _switch_to_blog(id)_. So this way we can create a shortcode that pulls up a template from the main site (in my case ID 1):
> 
> ```
> function elem_main_template($atts) {
>     $template_id = esc_attr($atts['id']);
>     
>     switch_to_blog(1); // your main site ID
>     $output = do_shortcode('[elementor-template id="' . $template_id . '"]');
>     restore_current_blog();
> 
>     return $output;
> }
> add_shortcode('main_template', 'elem_main_template');
> ```
> 
> Using a shortcode-widget and inserting: _[main_template id=3278]_ results in the widget from your 'main' site

**Comment by ofmarconi** [comment]
Will the global colors, fonts... be those of the sub-site?

**Comment by tjg37** [comment]
+1

**Comment by baker-industries** [comment]
+1

**Comment by GiannisTP** [comment]
+1 please

**Comment by kingfisher64** [comment]
@arielk - the initial request is nearly 8 years old, what are your thoughts on the actual request? Is this realistically going to be addressed? Thank you :)

**Comment by yabwee** [comment]
[elementor-shared-templates.zip](https://github.com/user-attachments/files/18046145/elementor-shared-templates.zip)

Here's a plugin I whipped up today that enables this functionality!

Notes:

-This is inspired by the shortcode provided by @cscheltinga, but with one crucial difference: his draws both the template structure AND content from the template's source site, which isn't ideal. Mine on the other hand, retrieves the template structure from the source site, but the content (including all dynamic content!) is populated from the current target site.

-This plugin does directly modify one Elementor file, adding a necessary do_action() in a strategic location. Though this is generally frowned upon, there's no other way (that I could find) to do it. In order to keep the modification despite Elementor being updated in the future, this plugin will automatically add the modification upon Elementor being updated. It's set to send a notification email to the admin when this happens.

In light of that, **after network installing this plugin, you need to trick Elementor into "updating" (which will trigger the modification) by editing plugins/elementor/elementor.php version # to less than the current version**. Example if it is currently "Version: 3.25.10" change it to "Version: 3.25.09" and then it will show up in the backend as needing to be updated.

-Finally, use the following shortcode to display a shared template:
**[est_template source_site_id=123 template_id=456]**

**Comment by kingfisher64** [comment]
Thanks @yabwee for sharing this. I'm going to set this up. Providing all goes well do you mind if I point Elementor support to this thread and your code to consider including in Elementor or using it in part? (assuming you haven't done already).

**Comment by yabwee** [comment]
> Thanks @yabwee for sharing this. I'm going to set this up. Providing all goes well do you mind if I point Elementor support to this thread and your code to consider including in Elementor or using it in part? (assuming you haven't done already).

Feel free to point them to this plugin's code. All they really need to do is add that action hook to plugins/elementor/core/base/document.php in do_print_elements(), which gives us access at that strategic point right before the content starts. 80% of the code in this plugin is just adding that one-liner in to their file!

**Comment by ofmarconi** [comment]
@yabwee Thank you! This looks really amazing! Congratulations

**Comment by maurocarreraok** [comment]
@arielk it would be nice if you can add this into the product since is label [on Jan 26, 2017]......

**Comment by kingfisher64** [comment]
@yabwee I have requested that the Elementor Developers look at what you have written, linking to this thread and your specific comment in a support ticket.

**Comment by kingfisher64** [comment]
Just one clarification @yabwee.

Would the source_site_id be 1 if it's the first site created in the multisite? If it could be different may I ask you or anyone else where that information would be found? 

I can see the template ID clearly under Theme Builder > Shortcode section.

Many thanks
 
> -Finally, use the following shortcode to display a shared template: **[est_template source_site_id=123 template_id=456]**

**Comment by yabwee** [comment]
To get any source site ID, just visit
https://{yourdomain.com}/wp-admin/network/sites.php
and hover on a site's Edit link. It will be in the URL: https://{
yourdomain.com}/wp-admin/network/site-info.php?*id=1*

**Comment by kingfisher64** [comment]
Thanks @yabwee I found this out just before you replied so removed the question. I also got a answer back from Elementor dev's as specifically requested. 

"I understand that you have questions regarding a Multisite Header footer template. While this feature has been requested before and added to the feature request queue, we cannot prioritize it upon request. The system is designed to prioritize requests based on their frequency, and the more requests we receive, the greater the likelihood that it will get to the queue for future consideration. I appreciate your understanding."

So if you want this to be included and are included in this thread - open a ticket up and ask them as otherwise not enough people are deemed to want it.

**Comment by yabwee** [comment]
> "I understand that you have questions regarding a Multisite Header footer template. While this feature has been requested before and added to the feature request queue, we cannot prioritize it upon request. **_The system is designed to prioritize requests based on their frequency, and the more requests we receive, the greater the likelihood that it will get to the queue for future consideration._** I appreciate your understanding."

Yeah, unfortunately we multisite users will always be in the minority, especially with requests this tricky. I resigned myself a while ago that if I want something done, I'll just have to rely on myself to (try to) get it done :)
